StRoot/StJetMaker/StJetMaker2012.cxx

```diff
@@ -152,31 +155,52 @@ int StJetMaker2012::Make()
 
 	// Get FMS towers                                                                                                 
 	StjTowerEnergyList fmsEnergyList;
-
 	if (jetbranch->anapars->useFms) {
 	  StjFMSMuDst fms;
 	  fmsEnergyList = fms.getEnergyList();
 	}
 
+    // Get FCS Towers
+          
+    StjTowerEnergyList fcsECalEnergyList; //FCS ECAL
+    if(jetbranch->anapars->useFcsECal){
+        StjFCSMuDst fcsECal;
+        fcsECal.useECAL();
+        fcsECalEnergyList = fcsECal.getEnergyList();
+        fcsECalEnergyList = jetbranch->anapars->FCSEcalemcCuts()(fcsECalEnergyList);
+
+    }
+    StjTowerEnergyList fcsHCalEnergyList; //FCS HCAL
+    if(jetbranch->anapars->useFcsHCal){
+        StjFCSMuDst fcsHCal;
+        fcsHCal.useHCAL();
+        fcsHCalEnergyList = fcsHCal.getEnergyList();
+        fcsHCalEnergyList = jetbranch->anapars->FCSHcalhcCuts()(fcsHCalEnergyList);
+
+    }
+
 	// Merge BEMC and EEMC towers
 	StjTowerEnergyList energyList;
 
 	copy(bemcEnergyList.begin(),bemcEnergyList.end(),back_inserter(energyList));
 	copy(eemcEnergyList.begin(),eemcEnergyList.end(),back_inserter(energyList));
 	copy(fmsEnergyList.begin(),fmsEnergyList.end(),back_inserter(energyList));
-
+    copy(fcsECalEnergyList.begin(),fcsECalEnergyList.end(),back_inserter(energyList));
+    copy(fcsHCalEnergyList.begin(),fcsHCalEnergyList.end(),back_inserter(energyList));
+          
 	// Apply hadronic correction to towers
 	energyList = jetbranch->anapars->correctTowerEnergyForTracks()(energyList,trackList);
 
 	// Convert tracks and towers to Lorentz vectors
 	FourList tpc4pList = StjeTrackListToStMuTrackFourVecList()(trackList);
```

Comment:
I think the track energy corrections to tower energies only apply to barrel and endcap, and maybe put this line after line copy(eemcEnergyList.begin(),eemcEnergyList.end(),back_inserter(energyList));

Do you have a similar correction method for forward tracks to FCS towers? If so you can leave it here and make sure the method is implemented.